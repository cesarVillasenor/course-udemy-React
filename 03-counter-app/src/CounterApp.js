import React, { useState } from 'react';
import PropTypes from  'prop-types';
//https://es.reactjs.org/docs/events.html

const CounterApp = ( { value = 10 } ) => {

    const [ counter, setCounter ] = useState( value );


    // handleAdd
    const handleAdd = (e) => {
        setCounter( counter + 1 );
        // setCounter( (c) => c + 1 )
    } 
    const handleReset = (e) => {
        setCounter( value );
    } 
    const handleSubstratct = (e) => {
        setCounter( counter - 1 );
    } 

    return <>
        <h1>Counter APP</h1>
        <h2> { counter } </h2>

        <button onClick={ handleAdd }> +1 </button>
        <button onClick={ handleReset }> reset </button>
        <button onClick={ handleSubstratct }> -1 </button>
    </>
};

CounterApp.propTypes = {
    counter: PropTypes.number
}

export default CounterApp;