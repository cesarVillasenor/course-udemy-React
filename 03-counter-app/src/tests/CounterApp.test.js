import React from 'react';
import { shallow } from 'enzyme';
import CounterApp  from '../CounterApp';

// task
// 1. Create the follwing tests for <CounterApp />
//      * Should show <CounterApp /> correctly (do match with snapShot)

//      * should show value by default of 100
//      * Use wrapper.find, takin the element html where it shown of counter value

describe('Tests of <CounterApp /> ', () => {
    
    let wrapper = shallow( <CounterApp /> );

    beforeEach( () => {

        wrapper = shallow( <CounterApp /> );

    })
    
    test('should show <CounterApp /> correctly (do match with snapShot)', () => {
        
        expect( wrapper ).toMatchSnapshot();
        
    });


    test('should show value by default of 100', () => {
        
        // const counter = '100';
        const wrapper = shallow( <CounterApp value={ 100 }/> );
      
        const h2Counter = wrapper.find('h2').text().trim();
        expect( h2Counter ).toBe( '100' )
        
    })
    
    test('should increase with button +1', () => {

        wrapper.find('button').at(0).simulate('click');

        const h2Counter = wrapper.find('h2').text().trim();

        expect( h2Counter ).toBe('11');
        
    })
    
        
    test('should decrease with button -1', () => {

        wrapper.find('button').at(2).simulate('click');

        const h2Counter = wrapper.find('h2').text().trim();

        expect( h2Counter ).toBe('9');
        
    })

    test('should reset value with button reset', () => {

        const wrapper = shallow( <CounterApp value={ 105 }/> );
      
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const h2Counter = wrapper.find('h2').text().trim();

        expect( h2Counter ).toBe('105');
        
    })

});