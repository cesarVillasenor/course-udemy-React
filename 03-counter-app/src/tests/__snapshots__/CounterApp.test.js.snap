// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <CounterApp />  should show <CounterApp /> correctly (do match with snapShot) 1`] = `
<Fragment>
  <h1>
    Counter APP
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     reset 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
